pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -93,19 +113,25 @@ def _deserialize_parameter(
     )
 
     if expression in UNARY_OPERATORS:
-        return UNARY_OPERATORS[expression](
-            _deserialize_parameter(param["lhs"], vars)
+        return cast(
+            ExpReturnType,
+            UNARY_OPERATORS[expression](
+                _deserialize_parameter(param["lhs"], vars)
+            ),
         )
     elif expression in BINARY_OPERATORS:
-        return BINARY_OPERATORS[expression](
-            _deserialize_parameter(param["lhs"], vars),
-            _deserialize_parameter(param["rhs"], vars),
+        return cast(
+            ExpReturnType,
+            BINARY_OPERATORS[expression](
+                _deserialize_parameter(param["lhs"], vars),
+                _deserialize_parameter(param["rhs"], vars),
+            ),
         )
     else:
-        raise AbstractReprError(f"Expression {param[expression]} invalid")
+        raise AbstractReprError(f"Expression '{param['expression']}' invalid.")
 
 
-def _deserialize_waveform(obj: Dict, vars: Dict) -> Waveform:
+def _deserialize_waveform(obj: dict, vars: dict) -> Waveform:
```

Comment:
nit: What's the rationale behind using `dict` instead of `typing.Dict` ?

Reply:
Personally, I prefer it because it avoids one import, but I enforce it because it is deprecated from Python 3.9 (but also when `annotations` is imported from `__future__`): https://docs.python.org/3/library/typing.html#typing.Dict